src/main/scala/devices/gpio/GPIO.scala

```diff
@@ -54,6 +57,8 @@ abstract class GPIO(busWidthBytes: Int, c: GPIOParams)(implicit p: Parameters)
         beatBytes = busWidthBytes),
       new GPIOPortIO(c))
     with HasInterruptSources {
+  val iofNode = BundleBridgeSource(() => new IOFPortIO(c))
+  val iofPort = InModuleBody { iofNode.bundle }
```

Comment:
it doesn't make a ton of sense to have these unconditionally here now if they may be empty. What do you think of the following change:

```suggestion
  val iofNodeOpt = c.includeIOF.option(BundleBridgeSource(() => new IOFPortIO(c)))
  val iofPortOpt = iofNodeOpt.map {i => InModuleBody { i.bundle}
```
and then get rid of the optional IOF inside the IOFPortIO?

Reply:
And then all the `if c.includeIOF{...}` below could just be `iofPortOpt.map{...}`

Reply:
It makes sense.